src/lib.rs

```diff
@@ -168,6 +168,7 @@ impl AllInstantiatedPerFileRules {
             .or_insert_with(|| {
                 instantiated_rule
                     .rule_instance
+                    .clone()
```

Comment:
These are because I changed the signature of `.instantiate_per_file()` to take a `self: Arc<Self>` parameter

The general thing that was a part of was wiring up the ability to get from the `RuleInstancePerFile` -> corresponding `RuleInstance` -> corresponding `Rule` (which generally seems like a desirable thing to be able to do and went along with needing to do something similar (but not type-erased) for doing the "magic `self.whatever` re-writings")